libs/cardano-ledger-core/testlib/Test/Cardano/Ledger/Core/KeyPair.hs

```diff
@@ -173,13 +181,18 @@ mkKeyHash :: Crypto c => Int -> KeyHash kd c
 mkKeyHash = hashKey . vKey . mkKeyPair
 
 mkKeyPair :: Crypto c => Int -> KeyPair r c
-mkKeyPair seed = KeyPair vk sk
+mkKeyPair = mkKeyPairWithSeed . Plain.serialize'
+
+mkKeyPairWithSeed :: forall r c. Crypto c => BS.ByteString -> KeyPair r c
+mkKeyPairWithSeed inputSeed = KeyPair vk sk
   where
+    seedSize = fromIntegral (DSIGN.seedSizeDSIGN (Proxy @(DSIGN c)))
+    seed
+      | BS.length inputSeed /= seedSize =
+          hashToBytes $ Hash.hashWith @Hash.Blake2b_256 id inputSeed
+      | otherwise = inputSeed
```

Comment:
This should probably be extracted into a separate function as it's also used in `mkBootKeyPairWithSeed`

Reply:
Oh yeah. Good point. They were added at different times, so I didn't notice the logic is exactly the same

Reply:
Done